content/en/documentation/for-contributors/control-plane/service-layers.md

```diff
@@ -142,6 +146,43 @@ This registers the request filter for the web context, and registers the authent
 filter. That way, whenever a HTTP request hits the `"yourcontext"` servlet container, the request filter gets invoked,
 delegating to the `SuperCustomAuthService` instance.
 
+#### 1.4.2 Authentication configuration
+
+The `auth-configuration` extension provides a way to configure available `AuthenticationService`s and bind them to a web context. To be compliant with the auth configuration, each `AuthenticationService` implementors  should register an `ApiAuthenticationProvider` in the `ApiAuthenticationProviderRegistry`. Each `ApiAuthenticationProvider` it's associated to an authentication type in the provider registry, and should create an instance of `AuthenticationService` based on the input `Config`.
+
+
+```java
+@Inject
+private ApiAuthenticationProviderRegistry providerRegistry;
+
+@Override
+public void initialize(ServiceExtensionContext context) {
+    // check the input config and build the `SuperCustomAuthService`
+    providerRegistry.register("customauthtype", (config) -> Result.success(new SuperCustomAuthService()));
+}
+```
+
+where the config is an object `Config` containing all the properties in the `auth` object `web.http.*.auth.*`
+
+For example applying the new `customauthtype` to the `management` api context a configuration would look like this:
+
+```
+web.http.management.auth.type=customauthtype
+web.http.management.auth.custom-property=custom
+```
+
+Currently available types are:
```

Comment:
This section about available types and how to configure could be put in the "for-adopters", as it is an useful information for those who just build and configure a runtime, while contributors should be aware about classes, interfaces and so on